packages/jupyter-ai/jupyter_ai/handlers.py

```diff
@@ -320,11 +321,11 @@ def get(self):
     @web.authenticated
     def post(self):
         try:
-            config = GlobalConfig(**self.get_json_body())
-            self.config_manager.update(config)
+            config = UpdateConfigRequest(**self.get_json_body())
+            self.config_manager.update_config(config)
             self.set_status(204)
             self.finish()
-        except ValidationError as e:
+        except (ValidationError, WriteConflictError, KeyEmptyError) as e:
```

Comment:
Not a blocker for this PR, but in the future, I'd love to see `WriteConflictError` handled by giving the user the choice to forcibly save the config. If the "force" option is set, we don't check for write conflicts.